docs/user-guide/reduction-workflow-guidelines.md

```diff
@@ -43,6 +40,80 @@ We plan to include the following in future versions of the guidelines:
 
 - *Provider*: A callable step in a workflow writing with Sciline.
 
+## C: Convention
+
+### C.1: Use common names and types
+
+**Reason**
+Helps with sticking to established practices and working across packages.
+
+**Table**
+Names use glob syntax, i.e., '*Filename' is any string that ends in 'Filename'.
+
+| Name                        | Type        | Description                                                               |
+|-----------------------------|-------------|---------------------------------------------------------------------------|
+| --- **Files** ---           |             |                                                                           |
+| Filename \| *Filename       | str         | Simple name of a file, must be processed into FilePath                    |
+| FilePath \| *FilePath       | Path        | Concrete path to a file on the host filesystem, ideally absolute          |
+| --- **Flags** ---           |             |                                                                           |
+| UncertaintyBroadcastMode    | enum        | E.g., `Enum('UncertaintyBroadcastMode', ['drop', 'upper_bound', 'fail'])` |
+| ReturnEvents                | bool        | Select whether to return events or histograms from the workflow           |
+| CorrectForGravity           | bool        | Toggle gravity correction                                                 |
+| --- **Misc** ---            |             |                                                                           |
+| NeXus*                      | Any         | Spelling of all NeXus-related keys                                        |
+| WavelengthBins \| *Bins     | sc.Variable | Bin-edges                                                                 |
+| RunTitle                    | str         | Extracted from NeXus or provided by user, can be used to find files       |
+
+### C.2: Use common names for generics
+
+**Reason**
+Helps with sticking to established practices and working across packages.
+
+**Note**
+If a workflow uses generics to parametrize its types, e.g., `Filename`,
+it should define new types used as tags and type vars constrained to those tags.
+
+**Table**
+
+| Name                                      | Type    | Description                                                     |
+|-------------------------------------------|---------|-----------------------------------------------------------------|
+| --- **Run IDs** ---                       |         |                                                                 |
+| SampleRun, BackgroundRun, ...             | Any     | Identifier for a run, only used as a type tag                   |
+| RunType                                   | TypeVar | Constrained to the run types used by the package, see above     |
```

Comment:
So is the rule to end typevars with `Type`?

Reply:
Don't know in general. I've only seen those two in practice.

Reply:
@SimonHeybrock Do we need to do anything about this?

Reply:
Should we make this a rule? I think it is quite confusing to know what is a typevar and what is and ID otherwise?